spacy_llm/tasks/rel.py

```diff
@@ -0,0 +1,114 @@
+from typing import Callable, Dict, Iterable, List, Optional, Union
+
+import jinja2
+from pydantic import BaseModel, Field, ValidationError, validator
+from spacy.tokens import Doc
+from wasabi import msg
+
+from ..registry import lowercase_normalizer, registry
+from .templates import read_template
+
+
+class RelationItem(BaseModel):
+    dep: int
+    dest: int
+    relation: str
+
+    @validator("dep", "dest", pre=True)
+    def clean_ent(cls, value):
+        if isinstance(value, str):
+            value = value.strip("ENT")
+        return value
+
+
+class EntityItem(BaseModel):
+    start_char: int
+    end_char: int
+    label_: str = Field(alias="label")
+
+
+class RELExample(BaseModel):
+    text: str
+    ents: List[EntityItem]
+    relations: List[RelationItem]
+
+
+def _preannotate(doc: Union[Doc, RELExample]) -> str:
+    """Creates a text version of the document with annotated entities."""
+    offset = 0
+
+    text = doc.text
+
+    for i, ent in enumerate(doc.ents):
+        end = ent.end_char
+        before, after = text[: end + offset], text[end + offset :]
+
+        annotation = f"[ENT{i}:{ent.label_}]"
+        offset += len(annotation)
+
+        text = f"{before}{annotation}{after}"
+
+    return text
+
+
+@registry.llm_tasks("spacy.REL.v1")
+class RELTask:
+    """Simple REL task. Populates a `Doc._.rel` custom attribute."""
+
+    _TEMPLATE_STR = read_template("rel")
+
+    def __init__(
+        self,
+        labels: str,
+        label_definitions: Optional[Dict[str, str]] = None,
+        examples: Optional[Callable[[], Iterable[Dict]]] = None,
+        normalizer: Optional[Callable[[str], str]] = None,
+        verbose: bool = False,
+    ):
+
+        if not Doc.has_extension("rel"):
+            Doc.set_extension("rel", default=[])
```

Comment:
This would need to be set in `__call__` and `pipe` instead for the pipeline component.

Reply:
Ah right, it wont get executed on deserialization?

Reply:
The problem isn't deserialization, it's multiprocessing.

Reply:
Thanks. Done.